tests/config_test.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for perfkitbenchmarker.configs."""
+
+import unittest
+
+from perfkitbenchmarker import benchmarks
+from perfkitbenchmarker import configs
+from perfkitbenchmarker import errors
+from perfkitbenchmarker import windows_benchmarks
+
+CONFIG_NAME = 'a'
+INVALID_NAME = 'b'
+INVALID_YAML_CONFIG = """
+a:
+  vm_groups:
+    default:
+    :
+"""
+VALID_CONFIG = """
+a:
+  vm_groups:
+    default:
+      vm_spec: *default_single_core
+"""
+CONFIG_A = """
+a:
+  flags:
+    flag1: old_value
+"""
+CONFIG_B = """
+a:
+  flags:
+    flag1: new_value
+"""
+REF_CONFIG = """
+b: &b
+  flags:
+    flag1: value
+
+a: *b
+"""
+
+
+class ConfigsTestCase(unittest.TestCase):
```

Comment:
Please add more tests - it doesn't look like there are any for the API beyond `LoadConfig`.
What happens if `flags` contains multiple keys and a subset are overridden?